### PR TITLE
feat: Publish column info

### LINKED ISSUE #63: Provide the table object id for selected rows/columns
You store the object id as `fieldId` in `_Column`, however you provide no way to publicly access this value.

I would need to access to it, to keep track of which columns comes from which tables in a JOIN.
See the [sqljocky](https://pub.dartlang.org/packages/sqljocky) package for an example of providing this information.

### PR DESCRIPTION
**Pull request by drager**
Fixes #63.

### CODE REVIEW

File: lib/postgresql.dart [hunk 1]
```diff
@@ -157,6 +157,23 @@ abstract class Row {
   
   /// An unmodifiable map of column names and values. 
   Map toMap();
+
+  List<Column> toColumns();
+}
+
+@proxy
```
Comment (Fox32) [inline lib/postgresql.dart]: I wonder why a proxy annotation is used here? _Column  doesn't seems to use noSuchMethod.
A copy and paste mistake?

File: lib/src/postgresql_impl/query.dart [hunk 2]
```diff
@@ -106,6 +107,8 @@ class _Row implements Row {
   List toList() => new UnmodifiableListView(_columnValues);
 
   Map toMap() => new Map.fromIterables(_columnNames, _columnValues);
+
+  List<Column> toColumns() => new List.from(_columns);
```
Comment (Fox32) [inline lib/src/postgresql_impl/query.dart]: How about using a UnmodifiableListView?

File: lib/postgresql.dart (hunk 1 above)
Comment (drager) [inline lib/postgresql.dart]: Ah, thats right. My mistake, thanks!

File: lib/src/postgresql_impl/query.dart (hunk 2 above)
Comment (drager) [inline lib/src/postgresql_impl/query.dart]: Ah yes, great idea! Thanks.

**Comment by xxgreg** [comment]
Thanks very much for this. I should be able to get this merged and published tomorrow. I've got some comments for now.

Definitely agree with Fox32 about UnmodifiableListView. This class is now actually merged into core with a factory constructor on List: new List.unmodifiable(). But I'm not sure if that is in a stable release yet. So we probably have to do it the old way for now.

I was also thinking about the name toColumns(), since the columns don't actually have any row values in them, I think perhaps getColumns() is more appropriate. This is very un-darty, but the reason behind it - is we can't use a columns getter because this is more likely to conflict with a query result which had a column named "columns". I'm open to other ideas here.

I'll add some comments directly on the code too.

**Comment by xxgreg** [comment]
Good work. I've pushed this to pub in version 0.3.1. Let me know if you have any problems.

@Fox32 Oliver - thanks for helping with the review ;)

**Comment by drager** [comment]
Great! Thank you both for review!

**Pull request merged**